### ISSUE TITLE
adds ssh-agent integration like in keepassxc

### ISSUE THREAD

**Issue by dementor5** [keeweb/keeweb#1145]
**Is your feature request related to a problem? Please describe.**

Open and private keys may stored in keepass database record and adds to ssh-agent on demande and affermative answer in popup.
In ubuntu it  works after installing ssh-askpass-gnome.
https://keepassxc.org/docs/#faq-ssh-agent-how

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by antelle** [comment]
Should be done as a plugin.

**Comment by dominikheinz** [comment]
Any progress on this? Looking forward for this feature.

**Comment by Syndlex** [comment]
I started on this plugin. 
https://gitlab.com/Syndlex/keeweb-ssh-agent
I had to use webpack to bundle the npm dependencies.

The problem is now that Keeweb can not find the uninstall method of the plugin. 
https://github.com/keeweb/keeweb/blob/64c41785f6bbd19076a8fa82cd6193294494a500/app/scripts/plugins/plugin.js#L370
@antelle can you give me a tip how we can go around this problem?
Without the npm dependencies it will get hart ever implementing this plugin.

**Comment by antelle** [comment]
> The problem is now that Keeweb can not find the uninstall method of the plugin.

I see that `uninstall` method is empty: https://gitlab.com/Syndlex/keeweb-ssh-agent/blob/master/plugin.js#L74 - this is what should be called on uninstall, isn't it?

**Comment by Syndlex** [comment]
The Problem is happening after i back everything with webpack.
When I remove all external dependencies and just use webpack everything works normal.

**Comment by antelle** [comment]
I guess, webpack removes the method and you need to fix your webpack configuration

**Comment by Syndlex** [comment]
Good and bad news.
Good: Webpack is working now. 

For future reference  I added:  `libraryTarget: 'commonjs2'` in the output config. 

bad news: Not every Node core module can be used in browser. The SSH-Agents libs I found all need a socket from the node `net` module. I didn't found a browser implementation. :/

**Comment by antelle** [comment]
We don't support native modules. For now there's no solution for that.

**Comment by antelle** [comment]
The bot was misconfigured, sorry for a wrong notification.

**Comment by cnmicha** [comment]
@Syndlex Is the current version of the plugin stable and ready to use?

**Comment by Syndlex** [comment]
@cnmicha unfortunately no.

Tldr chromium has to give socket access for this feature.

To make it clear. An ssh agent would need access to low level functionality. The underlying technology does not give access to these functions. There is nothing keeweb can do here. 

It could be possible with an extra deamon running on the System. Similar to the Browser extension. But at this point you could just use keewebxc.
